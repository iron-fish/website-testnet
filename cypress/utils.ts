export type TestLink = {
  isImage: boolean
  text: string
  href: string
}

export const linksMatch = (expected: TestLink[]) => {
  cy.get('a').should($links => {
    const hrefs = $links.map((_, el) => {
      const $el = Cypress.$(el)
      const isImage = $el.find('img').get().length > 0
      return {
        isImage,
        text: isImage ? $el.find('img').attr('alt') : $el.text(),
        href: $el.attr('href'),
      }
    })
    expect(hrefs.get()).to.deep.eq(expected)
  })
}

const getFont = (el: HTMLElement) =>
  (document && document.defaultView
    ? document.defaultView.getComputedStyle(el).getPropertyValue('font-family')
    : el && el.style && el.style.font
  )
    .split(/('.*?'|".*?"|\S+)/g)
    .map(z => z.replace(/,/g, ''))
    .filter(z => z.trim().length > 0)

export const fontsInUse = (expected: string[]) => {
  cy.get('*').should($elements => {
    const mapped = $elements
      .map((_, el) => getFont(el))
      .get()
      .reduce(
        (agg: string[], x: string) => (agg.includes(x) ? agg : agg.concat(x)),
        []
      )
    expect(mapped).to.deep.eq(expected)
  })
}
