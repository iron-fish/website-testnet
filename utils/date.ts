import { isBefore, isAfter, eachWeekOfInterval, addMinutes } from 'date-fns'
import { toDate, format, utcToZonedTime } from 'date-fns-tz'

import { ApiEvent } from 'apiClient'

// modified version of https://github.com/date-fns/date-fns/issues/1401#issuecomment-621897094
const makeRelativeConverter = () => {
  const now = new Date()
  const offset = now.getTimezoneOffset()
  if (offset === 0) return (x: Date) => x
  return (x: Date) => addMinutes(x, -1 * x.getTimezoneOffset())
}

export const weeksBetween = (start: Date, end: Date) => {
  const raw = eachWeekOfInterval({ start, end }, { weekStartsOn: 1 })
  const convert = makeRelativeConverter()
  return raw.map(convert)
}

export const eventsBetween = (
  start: Date,
  end: Date,
  events: ApiEvent[]
): ApiEvent[] =>
  events.filter(e => {
    const time = toDate(e.occurred_at)
    return isAfter(time, start) && isBefore(time, end)
  })

export const formatInTimeZone =
  (timeZone: string) => (date: Date, formatString: string) =>
    format(utcToZonedTime(date, timeZone), formatString, { timeZone })
export const formatUTC = formatInTimeZone('UTC')
export const formatEventDate = (d: Date) =>
  formatUTC(d, `y'-'MM'-'dd HH':'mm':'ss`)
