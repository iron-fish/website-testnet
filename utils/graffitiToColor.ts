const fishAvatarColors = [
  '#FFAFAF',
  '#94ED6B',
  '#A3E9FF',
  '#7657CE',
  '#E7B453',
  '#54FF17',
  '#1D4423',
  '#D46161',
  '#AFF1FF',
  '#F1CB00',
]

export function graffitiToColor(graffiti: string): string {
  // djb2 (xor)
  let hash = 5381

  for (let i = 0; i < graffiti.length; i++) {
    hash = (hash * 33) ^ graffiti.charCodeAt(i)
  }

  return fishAvatarColors[(hash >>> 0) % fishAvatarColors.length]
}
