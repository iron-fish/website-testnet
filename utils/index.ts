export * from './graffitiToColor'
export * from './numberToOrdinal'