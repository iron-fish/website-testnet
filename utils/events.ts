import { isBefore, isAfter } from 'date-fns'
import { toDate } from 'date-fns-tz'
import { formatUTC, weeksBetween } from './date'
import { ApiEvent } from 'apiClient'

export type WeeklyData = {
  week: number
  date: Date
  prior: Date
  events: ApiEvent[]
}

export const withinBounds = (start: Date, end: Date) => (e: ApiEvent) => {
  const time = toDate(e.occurred_at)
  return isAfter(time, start) && isBefore(time, end)
}
export const eventsBetween = (
  start: Date,
  end: Date,
  events: ApiEvent[]
): ApiEvent[] => events.filter(withinBounds(start, end))

export const getAllVisibleEvents = (weeklyData: WeeklyData[]) => {
  const m = new Map()
  const allEvents = weeklyData.flatMap(({ events }) => events)
  allEvents.forEach(e => m.set(e.id, e))
  return m
}

export const findMissingEvents = (
  allEvents: readonly ApiEvent[],
  weeklyData: WeeklyData[]
) => {
  const onlyMatched = getAllVisibleEvents(weeklyData)
  return allEvents.reduce((agg: ApiEvent[], x: ApiEvent) => {
    if (onlyMatched.has(x.id)) {
      return agg
    }
    return agg.concat(x)
  }, [])
}

const sortEventsByDate = (xs: ApiEvent[]) =>
  xs.sort((a: ApiEvent, b: ApiEvent): number =>
    a.occurred_at > b.occurred_at ? -1 : 1
  )
const makeCounter = () => {
  let x = 0
  return () => x++
}
export const bucketEvents = (
  start: Date,
  end: Date,
  rawEvents: readonly ApiEvent[]
): WeeklyData[] => {
  const weeks = weeksBetween(start, end)
  const counter = makeCounter()
  return weeks
    .reduce((agg: WeeklyData[], date: Date) => {
      const prev = agg[agg.length - 1]
      const prior = prev ? prev.date : start
      const events = sortEventsByDate(
        eventsBetween(prior, date, rawEvents as ApiEvent[])
      )
      return agg.concat({
        prior,
        date,
        events,
        week: counter(),
      })
    }, [])
    .reverse()
}

export const bucketEventsAndAccountForMissingOnes = (
  start: Date,
  end: Date,
  rawEvents: readonly ApiEvent[]
) => {
  const weeklyData = bucketEvents(start, end, rawEvents)
  const missingEvents = findMissingEvents(rawEvents, weeklyData)
  const eventsPriorToStart = missingEvents.filter(({ occurred_at: time }) =>
    isBefore(toDate(time), start)
  )
  if (eventsPriorToStart.length > 0) {
    return weeklyData.concat({
      prior: new Date(2020, 1, 1),
      date: start,
      events: eventsPriorToStart,
      week: -1,
    })
  }
  return weeklyData
}

export const formatEventDate = (d: Date) =>
  formatUTC(d, `y'-'MM'-'dd HH':'mm':'ss`)
