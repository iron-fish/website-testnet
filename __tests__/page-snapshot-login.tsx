import Login from 'pages/login'
import { autoSnapshot } from 'jest.utils'

it('renders Login page', autoSnapshot(Login))
