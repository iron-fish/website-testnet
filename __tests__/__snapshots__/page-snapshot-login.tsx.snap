// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login page 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col"
  >
    <nav
      class="font-extended relative hover:bg-white hover:shadow-navbar hover:text-black bg-ifpink text-black"
    >
      <div
        class="absolute z-50 h-screen w-screen bg-white text-black font-extended transition-all transform-gpu overflow-y-auto -translate-x-full pb-6 md:hidden"
      >
        <div
          class="flex flex-col px-5 max-w-xl mx-auto"
        >
          <div
            class="flex mt-7 mb-10 justify-between items-center"
          >
            <div>
              <svg
                fill="none"
                height="32"
                viewBox="0 0 165 18"
                width="190"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M69.33.1c-5.388 0-9.147 3.93-9.147 8.95 0 5.018 3.759 8.95 9.148 8.95 5.388 0 9.147-3.93 9.147-8.95S74.72.1 69.331.1zm.163 15.164c-2.2.057-3.51-2.013-3.617-6.083-.117-4.48 1.093-6.288 3.262-6.346 2.256-.06 3.531 1.713 3.647 6.165.108 4.096-1.065 6.206-3.292 6.264zM40.598.397h-7.12c-.395 0-.643.248-.643.644v.94c0 .42.147.569.518.716l.47.174c.519.197.692.47.692 1.013v10.333c0 .545-.173.815-.692 1.013l-.47.174c-.371.147-.518.296-.518.716v.94c0 .395.248.643.643.643h7.12c.395 0 .643-.248.643-.643v-.94c0-.42-.147-.57-.518-.716l-.47-.174c-.519-.198-.692-.47-.692-1.013V3.884c0-.545.173-.816.692-1.013l.47-.174c.371-.147.518-.296.518-.716v-.94c-.001-.397-.248-.644-.643-.644zM59.624 15.428l-.273-.148c-.667-.37-1.162-.643-1.803-1.656l-2.325-3.732c-.123-.198-.05-.347.174-.396 1.953-.518 3.065-1.656 3.065-3.684 0-3.83-2.987-5.415-9.364-5.415L43.43.446c-.395.05-.593.272-.593.643v.94c0 .42.147.57.47.692l.569.222c.47.198.643.446.643.94v10.334c0 .545-.173.815-.692 1.013l-.47.174c-.37.147-.518.296-.518.716v.94c0 .395.248.643.643.643h7.12c.42 0 .644-.248.644-.643v-.94c0-.42-.15-.57-.519-.716l-.47-.174c-.518-.198-.692-.47-.692-1.013v-3.831c0-.222.075-.297.248-.297.198 0 .321.148.495.446l3.387 6.131c.42.767.742 1.04 1.606 1.04h4.325c.42 0 .594-.223.594-.545v-.89c-.004-.4-.127-.57-.597-.843zM50.476 8.01h-.544a.338.338 0 01-.351-.35V2.938c0-.203.145-.35.35-.35h.572c1.923 0 3.114.697 3.114 2.63 0 1.934-.921 2.793-3.14 2.793zM163.807 2.869l.471-.174c.371-.147.518-.296.518-.716v-.94c0-.396-.222-.644-.643-.644h-7.12c-.396 0-.644.248-.644.644v.94c0 .42.148.569.519.716l.47.174c.518.197.692.47.692 1.013v3.115a.285.285 0 01-.297.296h-5.241a.285.285 0 01-.297-.296V3.882c0-.545.174-.816.692-1.013l.47-.174c.372-.147.519-.296.519-.716v-.94c0-.396-.222-.644-.644-.644h-7.119c-.396 0-.644.248-.644.644v.94c0 .42.147.569.519.716l.47.174c.518.197.692.47.692 1.013v10.333c0 .545-.174.815-.692 1.013l-.47.174c-.372.147-.519.296-.519.716v.94c0 .395.248.643.644.643h7.119c.42 0 .644-.248.644-.643v-.94c0-.42-.149-.569-.519-.716l-.47-.174c-.518-.198-.692-.47-.692-1.013v-4.103c0-.174.123-.297.297-.297h5.241c.174 0 .297.123.297.297v4.103c0 .545-.174.815-.692 1.013l-.47.174c-.371.147-.519.296-.519.716v.94c0 .395.248.643.644.643h7.12c.419 0 .643-.248.643-.643v-.94c0-.42-.147-.569-.518-.716l-.471-.174c-.518-.198-.692-.47-.692-1.013V3.882c0-.543.174-.816.692-1.013zM139.091 6.726l-3.016-.89c-1.483-.443-2.274-.964-2.274-1.83 0-.84.791-1.374 2.003-1.383 1.699-.013 3.216.766 5.391 3.115.296.32.617.445 1.063.197l.519-.296c.445-.246.513-.614.296-.989l-2.16-3.732c-.248-.44-.42-.743-.94-.743h-.239c-.593 0-.667.273-1.261.273-.766 0-1.589-.297-2.992-.297-3.881 0-6.403 1.878-6.403 5.191 0 2.768 2.003 4.326 4.475 5.143l3.214 1.063c1.581.519 2.423 1.114 2.423 2.201 0 1.162-.89 1.68-2.349 1.68-2.225 0-4.154-1.26-6.179-3.93-.321-.42-.644-.494-1.064-.296l-.815.395c-.371.174-.475.547-.246 1.013l2.276 4.673c.205.42.494.593.841.593h.295c.544 0 .914-.371 1.41-.371.914 0 1.83.446 3.881.446 4.104 0 7.021-1.905 7.021-5.563-.003-3.389-2.253-4.823-5.17-5.663zM126.573.397h-7.12c-.396 0-.644.248-.644.644v.94c0 .42.148.569.519.716l.47.174c.519.197.692.47.692 1.013v10.333c0 .545-.173.815-.692 1.013l-.47.174c-.371.147-.519.296-.519.716v.94c0 .395.248.643.644.643h7.12c.395 0 .643-.248.643-.643v-.94c0-.42-.147-.57-.518-.716l-.47-.174c-.519-.198-.692-.47-.692-1.013V3.884c0-.545.173-.816.692-1.013l.47-.174c.371-.147.518-.296.518-.716v-.94c-.002-.397-.248-.644-.643-.644zM115.834.47c-.164-.285-.321-.47-.692-.47h-.492c-.272 0-.424.063-.629.224-.224.171-.418.171-.903.171h-9.207c-.395 0-.643.249-.643.644v.94c0 .42.147.569.518.716l.47.174c.519.198.693.47.693 1.013v10.333c0 .545-.174.816-.693 1.013l-.47.174c-.371.147-.518.296-.518.716v.94c0 .395.248.644.643.644h7.467c.42 0 .644-.249.644-.644v-.94c0-.42-.123-.617-.519-.716l-.815-.222c-.545-.148-.692-.42-.692-.965V9.864c0-.174.123-.297.297-.297h.419c1.929 0 3.115.989 4.326 2.399.173.198.395.272.692.123l.617-.32c.273-.148.373-.387.248-.644l-1.915-3.932c-.11-.226-.247-.36-.53-.36h-.304c-.209 0-.341.056-.532.208a1.016 1.016 0 01-.643.228h-2.378a.285.285 0 01-.297-.297V3.81c0-.89.347-1.237 1.285-1.237 1.608 0 3.807 1.41 5.092 2.696.222.222.495.272.791.098l.668-.42c.222-.147.373-.382.222-.643L115.834.47zM96.514 15.626l-.57-.198c-.518-.174-.691-.47-.691-1.013V3.662c0-.545.173-.817.692-1.013l.47-.174c.371-.147.518-.296.518-.716v-.716c0-.396-.222-.644-.643-.644h-4.747c-.396 0-.644.248-.644.644v.716c0 .42.15.569.519.716l.47.174c.518.197.692.47.692 1.013v6.056a.184.184 0 01-.326.118L85.944.992c-.157-.238-.318-.397-.482-.475-.164-.079-.392-.118-.679-.118h-4.71c-.395 0-.643.248-.643.644v.94c0 .42.149.569.518.716l.47.174c.519.197.692.47.692 1.012V14.44c0 .545-.173.815-.692 1.013l-.47.174c-.371.147-.518.296-.518.716v.716c0 .395.248.644.643.644h4.747c.42 0 .644-.249.644-.644v-.716c0-.42-.147-.57-.519-.716l-.47-.174c-.518-.198-.692-.47-.692-1.013V7.623a.184.184 0 01.327-.117l6.987 9.602c.157.24.317.398.481.476.164.078.392.118.68.118h4.13c.42 0 .644-.249.644-.644v-.716c0-.42-.147-.593-.518-.716zM28.06 8.802L23.47.854a.633.633 0 00-.547-.317h-9.181a.633.633 0 00-.549.317L9.25 7.685a.285.285 0 01-.317.13.283.283 0 01-.21-.272V4.9A4.366 4.366 0 004.364.54H.634A.635.635 0 000 1.172v3.729a4.278 4.278 0 002.64 3.962.28.28 0 010 .518A4.263 4.263 0 000 13.34v3.728c0 .35.284.635.634.635h3.729a4.366 4.366 0 004.36-4.361v-2.643c0-.127.087-.24.21-.273a.287.287 0 01.317.13l3.943 6.831a.635.635 0 00.549.317h9.179a.633.633 0 00.548-.317l4.589-7.948a.628.628 0 00.084-.315.619.619 0 00-.082-.32zm-9.659 3.65c-1.79 0-3.247-1.495-3.247-3.333 0-1.837 1.456-3.333 3.247-3.333 1.79 0 3.247 1.496 3.247 3.333 0 1.838-1.456 3.334-3.247 3.334z"
                  fill="black"
                />
              </svg>
            </div>
            <button>
              <svg
                fill="none"
                height="26"
                width="27"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="fill-current"
                  d="M.86 23.334L24.196 0l1.945 1.944L2.805 25.28z"
                />
                <path
                  class="fill-current"
                  d="M26.139 23.334L2.804 0 .86 1.945l23.334 23.334z"
                />
              </svg>
            </button>
          </div>
          <a
            class="leading-relaxed text-4xl px-2 lg:px-3.5 3xl:px-5"
          >
            Get Started
          </a>
          <a
            class="leading-relaxed text-4xl px-2 lg:px-3.5 3xl:px-5"
          >
            Whitepaper
          </a>
          <a
            class="leading-relaxed text-4xl px-2 lg:px-3.5 3xl:px-5"
          >
            Roadmap
          </a>
          <button
            class="leading-relaxed text-4xl px-2 lg:px-3.5 3xl:px-5"
          >
            <span
              class="flex items-center justify-between h-full relative"
            >
              Company
              <span
                class="ml-2 transition-transform duration-500"
              >
                <svg
                  class="w-6 md:w-4"
                  viewBox="0 0 13 6"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 0L0 0L6.5 6L13 0Z"
                  />
                </svg>
              </span>
            </span>
          </button>
          <button
            class="leading-relaxed text-4xl px-2 lg:px-3.5 3xl:px-5"
          >
            <span
              class="flex items-center justify-between h-full relative"
            >
              Testnet
              <span
                class="ml-2 transition-transform duration-500"
              >
                <svg
                  class="w-6 md:w-4"
                  viewBox="0 0 13 6"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 0L0 0L6.5 6L13 0Z"
                  />
                </svg>
              </span>
            </span>
          </button>
          <div
            class="w-full h-12"
          />
          <button
            class="buttonRoot flex justify-center items-center font-extended rounded-full whitespace-nowrap transition border-black border-2 bg-transparent h-16 hover:bg-black hover:text-white px-6 py-3 relative text-2xl text-black text-center md:min-w-[8rem] md:h-12 md:ml-4 md:px-3 md:text-sm lg:min-w-[10rem] lg:px-6 lg:text-base"
          >
            <div
              class="userButton absolute flex flex-col top-[0.75rem] pointer-events-none md:pointer-events-auto md:top-[0.6rem] lg:top-[0.5rem]"
            >
              <div
                class="flex flex-row content-around justify-center md:justify-end lg:justify-center"
              >
                <div
                  class="userButtonSpan md:pt-[0.1rem] md:min-w-[5.1rem] md:w-[5.1rem] lg:w-[7rem] xl:max-w-[20rem] md:overflow-x-hidden md:overflow-y-visible md:truncate md:relative"
                >
                  <div
                    class="truncate relative max-w-[20rem]"
                  >
                    cooldev
                  </div>
                </div>
                <svg
                  fill="none"
                  height="24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41Z"
                  />
                </svg>
              </div>
              <a
                class="arrowed md:hidden md:opacity-0 z-50 px-6 py-3 mt-12 w-full m-auto bg-white text-black md:w-auto md:mx-0 md:rounded-full md:hover:bg-black md:hover:text-white md:border md:border-black"
              >
                Sign Out
                <div
                  class="hitState"
                />
              </a>
            </div>
          </button>
        </div>
      </div>
      <div
        class="transition-width delay-300 flex items-stretch justify-between m-auto max-w-[1700px] text-sm px-5 lg:text-[18px] 2lg:px-10 2lg:text-xl 1.5lg:px-16"
      >
        <div
          class="py-7"
        >
          <a>
            <svg
              fill="none"
              height="32"
              viewBox="0 0 165 18"
              width="190"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M69.33.1c-5.388 0-9.147 3.93-9.147 8.95 0 5.018 3.759 8.95 9.148 8.95 5.388 0 9.147-3.93 9.147-8.95S74.72.1 69.331.1zm.163 15.164c-2.2.057-3.51-2.013-3.617-6.083-.117-4.48 1.093-6.288 3.262-6.346 2.256-.06 3.531 1.713 3.647 6.165.108 4.096-1.065 6.206-3.292 6.264zM40.598.397h-7.12c-.395 0-.643.248-.643.644v.94c0 .42.147.569.518.716l.47.174c.519.197.692.47.692 1.013v10.333c0 .545-.173.815-.692 1.013l-.47.174c-.371.147-.518.296-.518.716v.94c0 .395.248.643.643.643h7.12c.395 0 .643-.248.643-.643v-.94c0-.42-.147-.57-.518-.716l-.47-.174c-.519-.198-.692-.47-.692-1.013V3.884c0-.545.173-.816.692-1.013l.47-.174c.371-.147.518-.296.518-.716v-.94c-.001-.397-.248-.644-.643-.644zM59.624 15.428l-.273-.148c-.667-.37-1.162-.643-1.803-1.656l-2.325-3.732c-.123-.198-.05-.347.174-.396 1.953-.518 3.065-1.656 3.065-3.684 0-3.83-2.987-5.415-9.364-5.415L43.43.446c-.395.05-.593.272-.593.643v.94c0 .42.147.57.47.692l.569.222c.47.198.643.446.643.94v10.334c0 .545-.173.815-.692 1.013l-.47.174c-.37.147-.518.296-.518.716v.94c0 .395.248.643.643.643h7.12c.42 0 .644-.248.644-.643v-.94c0-.42-.15-.57-.519-.716l-.47-.174c-.518-.198-.692-.47-.692-1.013v-3.831c0-.222.075-.297.248-.297.198 0 .321.148.495.446l3.387 6.131c.42.767.742 1.04 1.606 1.04h4.325c.42 0 .594-.223.594-.545v-.89c-.004-.4-.127-.57-.597-.843zM50.476 8.01h-.544a.338.338 0 01-.351-.35V2.938c0-.203.145-.35.35-.35h.572c1.923 0 3.114.697 3.114 2.63 0 1.934-.921 2.793-3.14 2.793zM163.807 2.869l.471-.174c.371-.147.518-.296.518-.716v-.94c0-.396-.222-.644-.643-.644h-7.12c-.396 0-.644.248-.644.644v.94c0 .42.148.569.519.716l.47.174c.518.197.692.47.692 1.013v3.115a.285.285 0 01-.297.296h-5.241a.285.285 0 01-.297-.296V3.882c0-.545.174-.816.692-1.013l.47-.174c.372-.147.519-.296.519-.716v-.94c0-.396-.222-.644-.644-.644h-7.119c-.396 0-.644.248-.644.644v.94c0 .42.147.569.519.716l.47.174c.518.197.692.47.692 1.013v10.333c0 .545-.174.815-.692 1.013l-.47.174c-.372.147-.519.296-.519.716v.94c0 .395.248.643.644.643h7.119c.42 0 .644-.248.644-.643v-.94c0-.42-.149-.569-.519-.716l-.47-.174c-.518-.198-.692-.47-.692-1.013v-4.103c0-.174.123-.297.297-.297h5.241c.174 0 .297.123.297.297v4.103c0 .545-.174.815-.692 1.013l-.47.174c-.371.147-.519.296-.519.716v.94c0 .395.248.643.644.643h7.12c.419 0 .643-.248.643-.643v-.94c0-.42-.147-.569-.518-.716l-.471-.174c-.518-.198-.692-.47-.692-1.013V3.882c0-.543.174-.816.692-1.013zM139.091 6.726l-3.016-.89c-1.483-.443-2.274-.964-2.274-1.83 0-.84.791-1.374 2.003-1.383 1.699-.013 3.216.766 5.391 3.115.296.32.617.445 1.063.197l.519-.296c.445-.246.513-.614.296-.989l-2.16-3.732c-.248-.44-.42-.743-.94-.743h-.239c-.593 0-.667.273-1.261.273-.766 0-1.589-.297-2.992-.297-3.881 0-6.403 1.878-6.403 5.191 0 2.768 2.003 4.326 4.475 5.143l3.214 1.063c1.581.519 2.423 1.114 2.423 2.201 0 1.162-.89 1.68-2.349 1.68-2.225 0-4.154-1.26-6.179-3.93-.321-.42-.644-.494-1.064-.296l-.815.395c-.371.174-.475.547-.246 1.013l2.276 4.673c.205.42.494.593.841.593h.295c.544 0 .914-.371 1.41-.371.914 0 1.83.446 3.881.446 4.104 0 7.021-1.905 7.021-5.563-.003-3.389-2.253-4.823-5.17-5.663zM126.573.397h-7.12c-.396 0-.644.248-.644.644v.94c0 .42.148.569.519.716l.47.174c.519.197.692.47.692 1.013v10.333c0 .545-.173.815-.692 1.013l-.47.174c-.371.147-.519.296-.519.716v.94c0 .395.248.643.644.643h7.12c.395 0 .643-.248.643-.643v-.94c0-.42-.147-.57-.518-.716l-.47-.174c-.519-.198-.692-.47-.692-1.013V3.884c0-.545.173-.816.692-1.013l.47-.174c.371-.147.518-.296.518-.716v-.94c-.002-.397-.248-.644-.643-.644zM115.834.47c-.164-.285-.321-.47-.692-.47h-.492c-.272 0-.424.063-.629.224-.224.171-.418.171-.903.171h-9.207c-.395 0-.643.249-.643.644v.94c0 .42.147.569.518.716l.47.174c.519.198.693.47.693 1.013v10.333c0 .545-.174.816-.693 1.013l-.47.174c-.371.147-.518.296-.518.716v.94c0 .395.248.644.643.644h7.467c.42 0 .644-.249.644-.644v-.94c0-.42-.123-.617-.519-.716l-.815-.222c-.545-.148-.692-.42-.692-.965V9.864c0-.174.123-.297.297-.297h.419c1.929 0 3.115.989 4.326 2.399.173.198.395.272.692.123l.617-.32c.273-.148.373-.387.248-.644l-1.915-3.932c-.11-.226-.247-.36-.53-.36h-.304c-.209 0-.341.056-.532.208a1.016 1.016 0 01-.643.228h-2.378a.285.285 0 01-.297-.297V3.81c0-.89.347-1.237 1.285-1.237 1.608 0 3.807 1.41 5.092 2.696.222.222.495.272.791.098l.668-.42c.222-.147.373-.382.222-.643L115.834.47zM96.514 15.626l-.57-.198c-.518-.174-.691-.47-.691-1.013V3.662c0-.545.173-.817.692-1.013l.47-.174c.371-.147.518-.296.518-.716v-.716c0-.396-.222-.644-.643-.644h-4.747c-.396 0-.644.248-.644.644v.716c0 .42.15.569.519.716l.47.174c.518.197.692.47.692 1.013v6.056a.184.184 0 01-.326.118L85.944.992c-.157-.238-.318-.397-.482-.475-.164-.079-.392-.118-.679-.118h-4.71c-.395 0-.643.248-.643.644v.94c0 .42.149.569.518.716l.47.174c.519.197.692.47.692 1.012V14.44c0 .545-.173.815-.692 1.013l-.47.174c-.371.147-.518.296-.518.716v.716c0 .395.248.644.643.644h4.747c.42 0 .644-.249.644-.644v-.716c0-.42-.147-.57-.519-.716l-.47-.174c-.518-.198-.692-.47-.692-1.013V7.623a.184.184 0 01.327-.117l6.987 9.602c.157.24.317.398.481.476.164.078.392.118.68.118h4.13c.42 0 .644-.249.644-.644v-.716c0-.42-.147-.593-.518-.716zM28.06 8.802L23.47.854a.633.633 0 00-.547-.317h-9.181a.633.633 0 00-.549.317L9.25 7.685a.285.285 0 01-.317.13.283.283 0 01-.21-.272V4.9A4.366 4.366 0 004.364.54H.634A.635.635 0 000 1.172v3.729a4.278 4.278 0 002.64 3.962.28.28 0 010 .518A4.263 4.263 0 000 13.34v3.728c0 .35.284.635.634.635h3.729a4.366 4.366 0 004.36-4.361v-2.643c0-.127.087-.24.21-.273a.287.287 0 01.317.13l3.943 6.831a.635.635 0 00.549.317h9.179a.633.633 0 00.548-.317l4.589-7.948a.628.628 0 00.084-.315.619.619 0 00-.082-.32zm-9.659 3.65c-1.79 0-3.247-1.495-3.247-3.333 0-1.837 1.456-3.333 3.247-3.333 1.79 0 3.247 1.496 3.247 3.333 0 1.838-1.456 3.334-3.247 3.334z"
                fill="black"
              />
            </svg>
          </a>
        </div>
        <div
          class="hidden md:flex items-center lg:text-xl"
        >
          <a
            class="h-full flex items-center whitespace-nowrap transition-font transition-fast transition-padding px-2 lg:px-3.5 3xl:px-5"
          >
            Get Started
          </a>
          <a
            class="h-full flex items-center whitespace-nowrap transition-font transition-fast transition-padding px-2 lg:px-3.5 3xl:px-5"
          >
            Whitepaper
          </a>
          <a
            class="h-full flex items-center whitespace-nowrap transition-font transition-fast transition-padding px-2 lg:px-3.5 3xl:px-5"
          >
            Roadmap
          </a>
          <button
            class="h-full flex items-center whitespace-nowrap transition-font transition-fast transition-padding px-2 lg:px-3.5 3xl:px-5"
          >
            <span
              class="flex items-center justify-between h-full relative"
            >
              Company
              <span
                class="ml-2 transition-transform duration-500"
              >
                <svg
                  class="w-6 md:w-4"
                  viewBox="0 0 13 6"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 0L0 0L6.5 6L13 0Z"
                  />
                </svg>
              </span>
            </span>
          </button>
          <button
            class="h-full flex items-center whitespace-nowrap transition-font transition-fast transition-padding px-2 lg:px-3.5 3xl:px-5"
          >
            <span
              class="flex items-center justify-between h-full relative"
            >
              Testnet
              <span
                class="ml-2 transition-transform duration-500"
              >
                <svg
                  class="w-6 md:w-4"
                  viewBox="0 0 13 6"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 0L0 0L6.5 6L13 0Z"
                  />
                </svg>
              </span>
            </span>
          </button>
          <button
            class="buttonRoot flex justify-center items-center font-extended rounded-full whitespace-nowrap transition border-black border-2 bg-transparent h-16 hover:bg-black hover:text-white px-6 py-3 relative text-2xl text-black text-center md:min-w-[8rem] md:h-12 md:ml-4 md:px-3 md:text-sm lg:min-w-[10rem] lg:px-6 lg:text-base"
          >
            <div
              class="userButton absolute flex flex-col top-[0.75rem] pointer-events-none md:pointer-events-auto md:top-[0.6rem] lg:top-[0.5rem]"
            >
              <div
                class="flex flex-row content-around justify-center md:justify-end lg:justify-center"
              >
                <div
                  class="userButtonSpan md:pt-[0.1rem] md:min-w-[5.1rem] md:w-[5.1rem] lg:w-[7rem] xl:max-w-[20rem] md:overflow-x-hidden md:overflow-y-visible md:truncate md:relative"
                >
                  <div
                    class="truncate relative max-w-[20rem]"
                  >
                    cooldev
                  </div>
                </div>
                <svg
                  fill="none"
                  height="24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41Z"
                  />
                </svg>
              </div>
              <a
                class="arrowed md:hidden md:opacity-0 z-50 px-6 py-3 mt-12 w-full m-auto bg-white text-black md:w-auto md:mx-0 md:rounded-full md:hover:bg-black md:hover:text-white md:border md:border-black"
              >
                Sign Out
                <div
                  class="hitState"
                />
              </a>
            </div>
          </button>
        </div>
        <button
          class="md:hidden"
        >
          <svg
            focusable="false"
            height="30"
            role="img"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Menu
            </title>
            <path
              d="M4 7h22M4 15h22M4 23h22"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </nav>
    <div
      class="w-full fixed z-30 flex toast top top-0 topHidden"
    >
      <div
        class="Toast max-w-[16rem] mx-auto flex bg-black text-white text-base px-6 py-3.5"
      >
        <div
          class="Toast__message relative"
        />
      </div>
    </div>
    <main
      class="bg-ifpink flex-1 font-extended"
    >
      <div
        class="md:w-4/5 w-full my-6 max-w-section mx-auto transition-width"
      >
        <div
          class="relative mb-2.5 ml-2.5 mr-2.5 "
        >
          <div
            class="absolute border border-black top-2.5 -bottom-2.5 left-2.5 -right-2.5 bg-transparent"
          />
          <div
            class="relative border border-black bg-white"
          >
            <div
              class="flex justify-center"
            >
              <div
                class="flex flex-col items-center md:px-4 px-5"
              >
                <h1
                  class="text-4xl text-center mb-4 mt-16"
                >
                  Log in to the testnet.
                </h1>
                <div
                  class="font-favorit flex flex-col px-4 py-3 mt-4 w-full border rounded-plus border-solid h-16 max-h-16 max-w-md border-black  "
                >
                  <label
                    class="text-xs mb-px3"
                    for="email"
                  >
                    Email
                    <span
                      class="text-md text-gray-500"
                    >
                      *
                    </span>
                  </label>
                  <input
                    class=""
                    id="email"
                    type="text"
                    value=""
                  />
                </div>
                <button
                  class="flex justify-center items-center font-extended rounded-full whitespace-nowrap transition border-black border-2 bg-black text-white hover:bg-transparent hover:text-black w-full mt-8 max-w-md mb-2 text-lg md:text-xl p-3 md:py-5 md:px-4"
                >
                  Login
                </button>
                <div
                  class="text-center text-xl py-8"
                >
                  Don't yet have an account?
                   
                  <a
                    class="text-iflightblue"
                  >
                    Sign Up
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
